src/components/api-request.js

```diff
@@ -411,7 +411,8 @@ export default class ApiRequest extends LitElement {
       `;
 
     // For Loop - Main
-    const reqBody = requestBodyTypes.find(req => req.mimeType === this.selectedRequestBodyType);
+    const reqBody = requestBodyTypes.find(req => req.mimeType === this.selectedRequestBodyType) || requestBodyTypes[0];
```

Comment:
I'm not totally sure this is the right fix, the solution seems a bit strange. According to the problem, my guess would be we need to reset the selectedRequestBodyType on route change by watching the input parameters. Or more specifically we could watch `this.request_body` changes and the update the selected value.